### DISCUSSION TITLE
Error: HTTP POST is not supported for /api/auth/callback/credentials

### DISCUSSION THREAD

**Discussion by callumok2004** [nextauthjs/next-auth#2664] [Help] [answered]
### Question 💬

Code / Filetree
![image](https://user-images.githubusercontent.com/35779365/132113700-74395d82-c1e9-4bf7-8bc1-bd1e64dbc4d6.png)

When I attempt to log in, it gives me the error mentioned in the title
![image](https://user-images.githubusercontent.com/35779365/132113707-88dc0ab5-125a-4c8b-abfb-71308cc4d754.png)

### How to reproduce ☕️

I am unsure how someone else could reproduce this, nor do I see why its happening in the first place

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment, marked as answer]
Me neither on first look, but generally you don't need to fetch as you do on line 13 if you have an API route. try extracting your logic and import the actual method for handling the login process.

UPDATE: An idea. 💡. I *think* relative paths for fetch in the backend is not supported. Try prepending your url with `http://localhost:3000` (or better yet, import the login logic as I suggested above)

**Reply by callumok2004** [reply]
I have tried that too, no luck. I don't think its even reaching the authorize function as adding debug logs returns nothing.

**Reply by balazsorban44** [reply]
I think I see the problem! Your `[...nextauth].ts` file seems to be not properly nested inside `auth`.

**Reply by callumok2004** [reply]
Oh my, I feel so stupid right now! Thanks for the help :P
